activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -74,6 +74,12 @@ def options
 
         private
 
+        def id_to_index_map(ids)
```

Comment:
@jturkel not sure. I left it there because it seems like something that could viably be reused by other subclasses.